### PR TITLE
[WIP] Make backups easier to understand. 

### PR DESCRIPTION
**Pull request by auxesis**
**This PR is a Work in Progress. Please don't merge until I say I think it's in an OK state.**

I've refactored the backup script into functions. 

The intention is to make the logic easier to follow, so the operator doesn't need to know all the details of how the backups are performed. 

Currently the only logic change is the number of backups to retain (was previously 5, now set to 1, due to disk space problems per #1104)

### CODE REVIEW

File: provisioning/roles/backups/files/database-backup.sh [hunk 1]
```diff
@@ -1,19 +1,35 @@
...
-do
-  oldest=`ls -1rt $BACKUP_DIR | head -n 1`
-  rm -rf $BACKUP_DIR/$oldest
-  no_backups=`ls -1 $BACKUP_DIR | wc -l`
+
+setup_backup
+perform_backup
+count_backups
+
+while [ $backup_count -gt 1 ]; do
```
Comment (equivalentideas) [inline provisioning/roles/backups/files/database-backup.sh]: What's the significance of the change here from `-gt 5` to `-gt 1`?

**Comment by equivalentideas** [comment]
@auxesis I've raised the one query I had, but overall the named functions seem like a useful extraction 👍 I can understand what's going on here :)

File: provisioning/roles/backups/files/database-backup.sh (hunk 1 above)
Comment (auxesis) [reply provisioning/roles/backups/files/database-backup.sh]: Great question. 

Right now the script makes sure there's never more than 5 backups. 

The new script will make sure there's never more than 1 backup. 

This is done based on the fact that we're constantly deleting old backups, and there are never more than 2 present at any point. 

I reckon we could refactor the number into a `$max_backups` variable though.

Comment (equivalentideas) [reply provisioning/roles/backups/files/database-backup.sh]: Nice that makes sense. Would be good to mention in the PR description that it's not just a refactor ;-)

File: provisioning/roles/backups/files/database-backup.sh [hunk 2]
```diff
@@ -1,19 +1,36 @@
...
-  oldest=`ls -1rt $BACKUP_DIR | head -n 1`
-  rm -rf $BACKUP_DIR/$oldest
-  no_backups=`ls -1 $BACKUP_DIR | wc -l`
+MAX_BACKUPS=1
+
+setup_backup
+perform_backup
+count_backups
+
+while [ $backup_count -gt $MAX_BACKUPS ]; do
```
Comment (auxesis) [inline provisioning/roles/backups/files/database-backup.sh:38]: @equivalentideas let me know what you think of naming the variable.

Comment (equivalentideas) [reply provisioning/roles/backups/files/database-backup.sh:38]: All seem clear to me. Is there some convention for the all caps/small caps difference? I'm not familiar with shell scripting at all by the way :)

**Comment by mlandauer** [comment]
As there hasn't been any activity on this WIP PR in a while I'm closing this

**Pull request closed without merging**